cmd/pmm-managed/main.go

```diff
@@ -164,29 +166,80 @@ func makeRDSService(ctx context.Context, deps *rdsServiceDependencies) (*rds.Ser
 		Supervisor:    deps.supervisor,
 		PortsRegistry: ports.NewRegistry(10000, 10999, reserved),
 	}
-	rds, err := rds.NewService(&rdsConfig)
+	rdsService, err := rds.NewService(&rdsConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	err = deps.db.InTransaction(func(tx *reform.TX) error {
-		return rds.ApplyPrometheusConfiguration(ctx, tx.Querier)
+		return rdsService.ApplyPrometheusConfiguration(ctx, tx.Querier)
 	})
 	if err != nil {
 		return nil, err
 	}
 	err = deps.db.InTransaction(func(tx *reform.TX) error {
-		return rds.Restore(ctx, tx)
+		return rdsService.Restore(ctx, tx)
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	return rds, nil
+	return rdsService, nil
+}
+
+func makePostgreSQLService(ctx context.Context, deps *rdsServiceDependencies) (*postgresql.Service, error) {
+	// collect already reserved ports
+	rows, err := deps.db.Query("SELECT listen_port FROM agents")
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	defer rows.Close()
+
+	var reserved []uint16
+	for rows.Next() {
+		var port uint16
+		if err = rows.Scan(&port); err != nil {
+			return nil, errors.WithStack(err)
+		}
+		reserved = append(reserved, port)
+	}
+	if err = rows.Err(); err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	serviceConfig := postgresql.ServiceConfig{
+		PostgreSQLdExporterPath: *agentPostgreSQLExporterF,
+
+		DB:            deps.db,
+		Prometheus:    deps.prometheus,
+		QAN:           deps.qan,
+		Supervisor:    deps.supervisor,
+		PortsRegistry: ports.NewRegistry(10000, 10999, reserved),
```

Comment:
We must share a registry. Let's refactor reserved ports and registry creation in that PR.